src/Gemstone.Threading/InterprocessLock.cs

```diff
@@ -194,6 +221,43 @@ public static NamedSemaphore GetNamedSemaphore(bool perUser, int maximumCount =
+    /// <param name="semaphoreName">The generated hash name of the semaphore.</param>
+    /// <param name="maximumCount">The maximum number of requests for the semaphore that can be granted concurrently.</param>
+    /// <param name="initialCount">The initial number of requests for the semaphore that can be granted concurrently, or -1 to default to <paramref name="maximumCount"/>.</param>
+    /// <param name="global">Determines if semaphore should be marked as global; set value to <c>false</c> for local.</param>
+    /// <returns>A uniquely named inter-process <see cref="NamedSemaphore"/> specific to <paramref name="name"/>; <see cref="NamedSemaphore"/> is created if it does not exist.</returns>
+    /// <remarks>
+    /// <para>
+    /// This function uses a hash of the <paramref name="name"/> when creating the <see cref="NamedSemaphore"/>, not the actual <paramref name="name"/> - this way
+    /// restrictions on the <paramref name="name"/> length do not need to be a user concern. All processes needing an inter-process <see cref="NamedSemaphore"/> need
+    /// to use this same function to ensure access to the same <see cref="NamedSemaphore"/>.
+    /// </para>
+    /// <para>
+    /// The <paramref name="name"/> can be a string of any length (must not be empty, null or white space) and is not case-sensitive. All hashes of the
+    /// <paramref name="name"/> used to create the global <see cref="NamedSemaphore"/> are first converted to lower case.
+    /// </para>
+    /// <para>
+    /// On POSIX systems, the <see cref="NamedSemaphore"/> exhibits kernel persistence, meaning instances will remain active beyond the lifespan of the
+    /// creating process. Named semaphores must be explicitly removed by invoking <see cref="NamedSemaphore.Unlink"/> when they are no longer needed.
+    /// Kernel persistence necessitates careful design consideration regarding the responsibility for invoking <see cref="NamedSemaphore.Unlink"/>.
+    /// Since the common use case for named semaphores is across multiple applications, it is advisable for the last exiting process to handle the
+    /// cleanup. In cases where an application may crash before calling <see cref="NamedSemaphore.Unlink"/>, the semaphore persists in the system,
+    /// potentially leading to resource leakage. Implementations should include strategies to address and mitigate this risk.
+    /// </para>
+    /// </remarks>
+    /// <exception cref="ArgumentNullException">Argument <paramref name="name"/> cannot be empty, null or white space.</exception>
+    /// <exception cref="UnauthorizedAccessException">The named semaphore exists, but the user does not have the minimum needed security access rights to use it.</exception>
+    [MethodImpl(MethodImplOptions.Synchronized)]
+    public static NamedSemaphore GetNamedSemaphore(string name, out string semaphoreName, int maximumCount = 10, int initialCount = -1, bool global = true)
```

Comment:
Since this API more or less requires that you create the semaphore in order to retrieve the "true name" of the semaphore--and the API is clearly designed to hide those details about the "true name" of the semaphore--wouldn't it be better to capture the "true name" as a private member of the `NamedSemaphore` class and expose `Unlink()` as a public member instead of as a static method?

Reply:
This has been updated